validator/src/journal/publisher.rs

```diff
@@ -453,6 +457,8 @@ impl SyncBlockPublisher {
     }
 
     pub fn on_batch_received(&self, batch: Batch) {
+        self.consensus_notifier
```

Comment:
Should this only be called after the permission check?  The batch may not be added to the pending queue if that check fails.

Reply:
Since we never mark batches that fail the permission check as invalid, yes I believe you are correct. Otherwise batches may be sent to consensus but never marked invalid or committed.